CODE/vulnscan.py

```
@@ -0,0 +1,186 @@
+    mime_type, _ = mimetypes.guess_type(file_path)
+    if mime_type and mime_type.startswith('text'):
+        with open(file_path, 'r', encoding='utf-8', errors='ignore') as file:
+            content = file.read()
+        return is_sensitive(model, vectorizer, content)
+    else:
+        with open(file_path, 'r', errors='ignore') as file:
+            content = file.read()
+        return is_sensitive(model, vectorizer, content)
+
+
+def scan_directory(model: torch.nn.Module, vectorizer, dir_path: str) -> dict[str, tuple[bool, float]]:
```

Ruff (0.8.0)

Missing type annotation for function argument `vectorizer`

(ANN001)